enhancements/oc/mirroring-manifests.md

```diff
@@ -105,78 +98,104 @@ mirror
+It may also be the case that there are multiple clusters within the air gapped network.
+In such a case it makes sense to configure a central registry in the air gapped environment from which every cluster can pull upgrade imagery.
+Assuming the aforementioned use case and the use of removable media that will be sneakernetted across the gap, the steps for mirroring are:
+
+1. Connect the removable media to a system connected to the internet.
+2. Mirror the images and manifests to a directory on the removable media:
+	```console
+	$ oc adm release mirror --to-dir=<removable-media-path>/mirror quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+	```
+3. The enhanced `oc adm release mirror ...` command will output the path, similar to the example below, to a configmap created containing the image signature. Note this path as it will be used later when the configmap is applied to a cluster to be upgraded. 
+	```console
+	Configmap <removable-media-path>/mirror/manifests/sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92 created
+	...
+4. Run any required checks/scrubs on the removable media.
+5. Connect the removable media to a host within the air gapped environment that has access to both the central registry and to any cluster requiring upgrade.
+6. Upload the mirrored images to the central registry:
+	```console
+	$ oc image mirror --from-dir=<removable-media-path>/mirror file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+	```
+7. Use oc to login to a given cluster to be upgraded.
+8. Apply the the mirrored configmap, noted above, to the connected cluster for use by CVO for upgrade image verification:
+	```console
+	$ oc apply -f <removable-media-path>/mirror/manifests/sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+	```
+9. Perform the upgrade:
+	```console
+	$ oc adm upgrade --to-image REGISTRY/REPOSITORY/release@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+	```
 
 ### User Stories
 
 #### Release-signature ConfigMaps
 
-FIXME
+Currently a cluster upgrade can be accomplished on a cluster that does not have an active connection to the internet.
+However manual steps are required to create a ConfigMap containing the signature data required for update image verification.
+This enhancement will automatically create the ConfigMap from the image manifest files.
 
 #### In-cluster Cincinnati
 
 FIXME
 
 ### Risks and Mitigations
 
-FIXME
+One risk considered is that a user familiar with the current behavior of the `oc adm release mirror ...` command may inadvertently apply manifests to thier cluster.
+This has been addressed by requiring the explicit `--apply-manifests` argument to apply manifests to a cluster.
 
 ## Design Details
 
 ### Test Plan
 
-FIXME
+The current e2e test, that makes sure the command always exits successfully and that certain apsects of the content are always present, will be modified to validate the new functioanlity as well.
```

Comment:
nit: "functioanlity" -> "functionality".

Also, we have existing CI jobs that mirror into a separate repository, install from that mirror ([code][1]), and then attempt to update the cluster.  But that's currently not tied to actually black-holing network access (there is work in flight for *that* in openshift/release#5308, although that could be ported to the multi-step framework and probably simplified now that we have bring-your-own-VPC support).  So there are quite a few flavor posibilities:

* mirror to registry A, install cluster, mirror the update target to registry A, update (forcing, because we usually don't have signatures for CI builds?  Or do we?).
* mirror to registry A, blackhole access to the main CI registry, install cluster, mirror the update into the cluster itself, update.
* vanilla install from the CI registry, mirror the update into the cluster itself, update.
* vanilla install from the CI registry, mirror the update to disk, apply to the cluster itself with `oc`, update.
* etc.

Are we planning on adding some of those?

[1]: https://github.com/openshift/ci-tools/blob/993ab7400b665fda8618f83e4022db0ebd90ebcc/pkg/steps/clusterinstall/template.go#L408-L423

Reply:
I took that from another approved/merged oc enhancement so modified it to read as we'll write whatever new tests are needed to test the NEW functionality which of course assumes there are already e2e tests that test the current functionality. Maybe not but for the purposes of this enhancement we would should only need to speak to testing the enhancement itself - I would think.